packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -666,23 +715,33 @@ mixin _CupertinoSheetRouteTransitionMixin<T> on PageRoute<T> {
     Animation<double> secondaryAnimation,
     Widget child,
     bool enableDrag,
+    double topGap,
   ) {
     final bool linearTransition = route.popGestureInProgress;
     return CupertinoSheetTransition(
       primaryRouteAnimation: animation,
       secondaryRouteAnimation: secondaryAnimation,
       linearTransition: linearTransition,
+      topGap: topGap,
       child: _CupertinoDragGestureDetector<T>(
         enabledCallback: () => enableDrag,
-        onStartPopGesture: () => _startPopGesture<T>(route),
+        onStartPopGesture: () => _startPopGesture<T>(route, topGap),
         child: child,
       ),
     );
   }
 
+  @override
+  bool canTransitionFrom(TransitionRoute<dynamic> previousRoute) {
+    return !_hasCustomTopGap;
+  }
+
   @override
   bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
-    return nextRoute is _CupertinoSheetRouteTransitionMixin;
+    if (this is CupertinoSheetRoute<dynamic> && _hasCustomTopGap) {
+      return false;
+    }
+    return nextRoute is _CupertinoSheetRouteTransitionMixin && !_hasCustomTopGap;
```

Comment:
This check for !_hasCustomTopGap is redundant, because it already happens above.

Reply:
Sure, will do.